### PR TITLE
CORE-11211 Implemented Flow API for External Messaging

### PR DESCRIPTION
**Pull request by owenstanford**
Added the implementation of the send-only external messaging API + supporting services for reading metadata and validating topic existence.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -99,6 +99,7 @@ include 'components:crypto:crypto-softhsm-impl'
 include 'components:db:db-connection-manager'
 include 'components:db:db-connection-manager-impl'
 include 'components:domino-logic'
+include 'components:external-messaging-services'
```
Comment (filipesoliveira) [inline settings.gradle:102]: Small detail. Did you actually mean to write the component name in the plural? Other components have "service" in their name.

**Review by filipesoliveira — APPROVED** [review]
LGTM.

**Review by ronanbrowne — APPROVED** [review]
lgtm - build

**Review by JamesHR3 — COMMENTED** [review]
Only reviewed the flow classes, but this looks sensible from that perspective.

**Review by LWogan — COMMENTED** [review]
The flow engine changes look fine to me. I wont put individual comments on the PR but most of the APIs and classes are missing KDocs. Im keen have descriptive and in some cases verbose Kdocs on new stuff as theres a lot of new people to Flow Worker

**Pull request merged**